### ISSUE TITLE
Consider recognizing `0.0.0.0` as a local host address for various `bench-scripts` interfaces

### ISSUE THREAD

**Issue by portante** [distributed-system-analysis/pbench#2840]
Why can't somebody invoke `pbench-uperf` with the client specified as `0.0.0.0` and the server being some other host?

E.g. `pbench-uperf -r 10 -t stream -m 1 -p tcp -i 1 --samples 1 --clients 0.0.0.0 --servers 192.168.122.1`

**Comment by ldoktor** [comment]
Well this should work, only pbench will treat 0.0.0.0 as a remote address.

0.0.0.0 => listen on **all** interfaces

the main question here was, whether that wouldn't be a better default as that way we can use remote `--servers` without the need to modify the `--clients` as the port will be opened on all of them.

And note that for `--servers` we still need to use the `127.0.0.1` (or `localhost`) as it's not possible to connect to 0.0.0.0 (how would one connect to all local interfaces simultaneously).

**Comment by dbutenhof** [comment]
Yes ... and then there's `::` and `*` if we want to think about breaking into the 20th century!
